### PR TITLE
Upgrade to latest rocksdb

### PR DESCRIPTION
**Pull request by dvdplm**
Upgrades `kvdb-*` crates which brings with it an upgrade to rocksdb
~6.7.3~ 6.7.4.

### CODE REVIEW

File: ethcore/snapshot/src/service.rs [hunk 1]
```diff
@@ -753,7 +751,8 @@ impl<C> Service<C> where C: SnapshotClient + ChainInfo {
 
 	/// Feed a chunk with the Restoration
 	fn feed_chunk_with_restoration(&self, restoration: &mut Option<Restoration>, hash: H256, chunk: &[u8], is_state: bool) -> Result<(), Error> {
-		let (result, db) = {
+		// todo[dvdplm] can re-work this to avoid the db clone (if it's indeed correct)
```
Comment (ordian) [inline ethcore/snapshot/src/service.rs:754]: is this about `rest.db.clone`? it is an `Arc`, so cloning it is cheap (one atomic increment).

File: parity/db/rocksdb/blooms.rs [hunk 2]
```diff
@@ -63,7 +63,7 @@ pub fn migrate_blooms<P: AsRef<Path>>(path: P, config: &DatabaseConfig) -> Resul
 	// 1u8 -> TraceDBIndex::BloomGroups
 	// 0u8 -> level 0
 	let trace_blooms_iterator = db.key_value()
-		.iter_from_prefix(4, &[1u8, 0u8])
+		.iter_with_prefix(4, &[1u8, 0u8])
 		.filter(|(key, _)| key.len() == 6)
 		.take_while(|(key, _)| {
 			key[0] == 1u8 && key[1] == 0u8
```
Comment (ordian) [inline parity/db/rocksdb/blooms.rs:69]: the `take_while` can be removed here and above

File: ethcore/snapshot/src/lib_REMOTE_32285.rs [hunk 3]
```diff
@@ -0,0 +1,550 @@
+// Copyright 2015-2020 Parity Technologies (UK) Ltd.
```
Comment (ordian) [inline ethcore/snapshot/src/lib_REMOTE_32285.rs:1]: you didn't mean to commit them, right?

File: parity/db/rocksdb/blooms.rs (hunk 2 above)
Comment (dvdplm) [reply parity/db/rocksdb/blooms.rs:69]: Now that I think about it: why do we even need to keep this migration? There aren't any supported clients out there that don't have their trace blooms migrated already, are there?

Comment (ordian) [reply parity/db/rocksdb/blooms.rs:69]: Right, the migration was introduced 2.0, so we can assume that nobody runs 1.x anymore
https://ethernodes.org/client/parity-ethereum

Comment (dvdplm) [reply parity/db/rocksdb/blooms.rs:69]: Cool. Deleting.

**Comment by ordian** [comment]
any chance to test 6.8.1 https://github.com/rust-rocksdb/rust-rocksdb/pull/428 (not published yet)? it contains a potential fix for #11586

File: ethcore/snapshot/src/service.rs [hunk 4]
```diff
@@ -757,52 +753,36 @@ impl<C> Service<C> where C: SnapshotClient + ChainInfo {
...
-									drop(db);
-									return self.finalize_restoration(&mut *restoration);
-								},
-								false => Ok(())
-							}
-						}
-						Err(e) => Err(e)
-					};
-					(res, db)
+	fn feed_chunk_with_restoration(
```
Comment (adria0) [inline ethcore/snapshot/src/service.rs:756]: Thanks for making this code readable.

File: Cargo.toml [hunk 5]
```diff
@@ -59,9 +59,9 @@ parity-local-store = { path = "miner/local-store" }
 parity-path = "0.1"
 parity-rpc = { path = "rpc" }
 parity-runtime = "0.1.1"
-parity-secretstore = { git = "https://github.com/paritytech/secret-store", branch = "v1.x", optional = true }
+parity-secretstore = { git = "https://github.com/paritytech/secret-store", rev = "e041c80d6f073ca653fec9f653236a11cc20fb56", optional = true }
```
[the remark this answers is not in the document]
Comment (dvdplm) [reply Cargo.toml:62]: Good pt, fixing.

**Pull request merged**